core/foundation/src/RLogger.cxx

```diff
@@ -14,46 +14,101 @@
  *************************************************************************/
 
 #include "ROOT/RLogger.hxx"
-#include "ROOT/RMakeUnique.hxx"
-#include <iostream>
-#include <sstream>
 
+#include "ROOT/RMakeUnique.hxx"
 #include "TError.h"
 
+#include <array>
+#include <algorithm>
+
+using namespace ROOT::Experimental;
+
 // pin vtable
-ROOT::Experimental::RLogHandler::~RLogHandler() {}
+RLogHandler::~RLogHandler() {}
 
 namespace {
-class RLogHandlerDefault: public ROOT::Experimental::RLogHandler {
+class RLogHandlerDefault : public RLogHandler {
 public:
    // Returns false if further emission of this log entry should be suppressed.
-   bool Emit(const ROOT::Experimental::RLogEntry &entry) override;
+   bool Emit(const RLogEntry &entry) override;
 };
 
-bool RLogHandlerDefault::Emit(const ROOT::Experimental::RLogEntry &entry)
+bool RLogHandlerDefault::Emit(const RLogEntry &entry)
 {
-   constexpr static std::array<const char *, 5> sTag{{"Debug", "Info", "Warning", "Log", "FATAL"}};
+   constexpr static std::array<const char *, 5> sTag{{"Debug", "Info", "Warning", "Error", "FATAL"}};
    std::stringstream strm;
    strm << "ROOT ";
-   if (!entry.fGroup.empty())
-      strm << '[' << entry.fGroup << "] ";
+   auto channel = entry.fChannel;
+   if (channel && !channel->GetName().empty())
+      strm << '[' << channel->GetName() << "] ";
    strm << sTag[static_cast<int>(entry.fLevel)];
 
-   if (!entry.fFile.empty())
-      strm << " " << entry.fFile << ':' << entry.fLine;
-   if (!entry.fFuncName.empty())
-      strm << " in " << entry.fFuncName;
+   if (!entry.fLocation.fFile.empty())
+      strm << " " << entry.fLocation.fFile << ':' << entry.fLocation.fLine;
+   if (!entry.fLocation.fFuncName.empty())
+      strm << " in " << entry.fLocation.fFuncName;
 
    static constexpr const int errorLevelOld[] = {0, 1000, 2000, 3000, 6000};
```

Comment:
I don't think these are correct but I might be wrong: we have a `kUnset`, don't have a `kPrint`, `kFatal` is element `1` in our enum but iiuc it's 6000 in `errorLevelOld`...